zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -400,141 +348,213 @@ impl<F: Field, const MAX_VERIF: usize> SignVerifyChip<F, MAX_VERIF> {
         })
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn assign_rlc_le(
+        &self,
+        config: &SignVerifyConfig,
+        ctx: &mut RegionCtx<F>,
+        chips: &ChipsRef<F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+        name: &str,
+        q_rlc: Selector,
+        challenge: Value<F>,
+        inputs_le: impl IntoIterator<Item = Term<F>>,
+    ) -> Result<AssignedCell<F, F>, Error> {
+        let zero = chips.main_gate.assign_constant(ctx, F::zero())?;
```

Comment:
If instead of creating a new constant for every call to `assign_rlc_le` we do it once for the entire circuit, do you think we could have a non-negligible optimization benefit?

Reply:
> If instead of creating a new constant for every call to `assign_rlc_le` we do it once for the entire circuit

I agree so, but I feel it's ugly to implement in Tx circuit, it should be an API `constrain_constant` provided from `MainGate` just like what `Region` provides, and inside `MainGate` it tracks assigned constant cells and reuse to avoid such unnecessary cost.

I have created an feature request issue here https://github.com/privacy-scaling-explorations/halo2wrong/issues/47

Reply:
Handling the reuse of constants from the `MainGate` sounds great!